presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -191,6 +188,194 @@ This table can then be queried in Presto::
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:** This together with ``hive.metastore.partition-batch-size.min`` defines range of partition sizes read from Hive. First partition is always of size ``hive.metastore.partition-batch-size.min`` and each following partition is two times bigger then previous up to ``hive.mestastore.partition-batch-size.max`` (the formula for ``n`` partition size is min(``hive.metastore.partition-batch-size.max``, (``2``^``n``) * ``hive.metastore.partition-batch-size.min``)). This algorithm allows to adjust partition size live to what is required. If size of queries in system differs siginificantly, then this range should be extended to better adjust to processed case. In case of cluster working with queries with about the same size, both values may be same for maximal attunement giving slight edge in processing time.
+
+
+``hive.metastore.partition-batch-size.min``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``10``
+ * **Description:** See ``hive.metastore.partition-batch-size.max``.
+
+
+``hive.optimized-reader.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** *Deprecated* Enables number of reader improvements introduced by alternative ORC implementation. The new reader supports vectorized reads, lazy loading, and predicate push down, all of which make the reader more efficient and typically reduces wall clock time for a query. However as the code has changed significantly it may or may not introduce some minor issues, so it can be disabled if some  problems with environment are noticed.
+
+
+``hive.orc.max-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``8 MB``
+ * **Description:** Serves as default value for ``orc_max_buffer_size`` session properties defining max size of ORC read operators. Higher value will allow bigger chunks to be processed but will decrease concurrency level.
+
+
+``hive.orc.max-merge-distance``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``1 MB``
+ * **Description:** Serves as default value for ``orc_max_merge_distance`` session property. Defines maximum size of gap between two reads to merge into a single read. The reads may be merged if distance between requested data ranges in data source is smaller or equal to this value.
+
+
+``hive.orc.stream-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``8 MB``
+ * **Description:** Serves as default value for ``orc_stream_buffer_size`` session property defining max size of ORC streaming operators. Higher value allows to load and process bigger chunks of data possibly increasing latency and decreasing parallelism but also increasing processing efficiency.
```

Comment:
Serves as the default value for the ``orc_stream_buffer_size`` session property.   It defines the maximum size of ORC streaming operators.  A higher value allows for loading and processing larger chunks of data.  It will increase processing efficiency, but may also increase query latency and decrease parallelism.

what does max size of ORC streaming operators mean?  Is that the memory allocated to those java processes?